SmartDeviceLinkTests/SDLSubscribeButtonManagerSpec.m

```diff
@@ -0,0 +1,716 @@
+//
+//  SDLSubscribeButtonManagerSpec.m
+//  SmartDeviceLinkTests
+//
+//  Created by Nicole on 5/22/20.
+//  Copyright © 2020 smartdevicelink. All rights reserved.
+//
+
+#import <Quick/Quick.h>
+#import <Nimble/Nimble.h>
+#import <OCMock/OCMock.h>
+
+#import "SDLError.h"
+#import "SDLOnButtonEvent.h"
+#import "SDLOnButtonPress.h"
+#import "SDLRPCNotificationNotification.h"
+#import "SDLSubscribeButtonManager.h"
+#import "SDLSubscribeButtonObserver.h"
+#import "SDLSubscribeButtonResponse.h"
+#import "SDLSubscribebutton.h"
+#import "SDLUnsubscribeButton.h"
+#import "SDLUnsubscribeButtonResponse.h"
+#import "TestConnectionManager.h"
+#import "TestSubscribeButtonObserver.h"
+
+@interface SDLSubscribeButtonManager()
+
+@property (weak, nonatomic) id<SDLConnectionManagerType> connectionManager;
+@property (strong, nonatomic) NSMutableDictionary<SDLButtonName, NSMutableArray<SDLSubscribeButtonObserver *> *> *subscribeButtonObservers;
+
+@end
+
+QuickSpecBegin(SDLSubscribeButtonManagerSpec)
+
+describe(@"subscribe button manager", ^{
+    __block SDLSubscribeButtonManager *testManager = nil;
+    __block TestConnectionManager *testConnectionManager = nil;
+
+    __block SDLSubscribeButtonResponse *testSubscribeSuccessResponse = nil;
+    __block SDLSubscribeButtonResponse *testSubscribeFailureResponse = nil;
+    __block SDLUnsubscribeButtonResponse *testUnSubscribeSuccessResponse = nil;
+    __block SDLUnsubscribeButtonResponse *testUnSubscribeFailureResponse = nil;
+
+    beforeEach(^{
+        testSubscribeSuccessResponse = [[SDLSubscribeButtonResponse alloc] init];
+        testSubscribeSuccessResponse.success = @YES;
+        testUnSubscribeSuccessResponse.resultCode = SDLResultSuccess;
+
+        testSubscribeFailureResponse = [[SDLSubscribeButtonResponse alloc] init];
+        testSubscribeFailureResponse.success = @NO;
+        testUnSubscribeSuccessResponse.resultCode = SDLResultDisallowed;
+
+        testUnSubscribeSuccessResponse = [[SDLUnsubscribeButtonResponse alloc] init];
+        testUnSubscribeSuccessResponse.success = @YES;
+        testUnSubscribeSuccessResponse.resultCode = SDLResultSuccess;
+
+        testUnSubscribeFailureResponse = [[SDLUnsubscribeButtonResponse alloc] init];
+        testUnSubscribeFailureResponse.success = @NO;
+        testUnSubscribeFailureResponse.resultCode = SDLResultDisallowed;
+    });
+
+    context(@"lifecycle", ^{
+        beforeEach(^{
+            testConnectionManager = [[TestConnectionManager alloc] init];
+            testManager = [[SDLSubscribeButtonManager alloc] initWithConnectionManager:testConnectionManager];
+        });
```

Comment:
I think that this can be moved to the top `beforeEach` and the tests can be at that level and this `context` can be removed

Reply:
Fixed. Removed.